src/main/java/com/mraof/minestuck/entity/ServerCursorEntity.java

```diff
@@ -0,0 +1,235 @@
+package com.mraof.minestuck.entity;
+
+import com.mraof.minestuck.MinestuckConfig;
+import com.mraof.minestuck.network.MSPacketHandler;
+import com.mraof.minestuck.network.ServerCursorPacket;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.Tag;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.network.protocol.Packet;
+import net.minecraft.world.entity.*;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.AABB;
+import net.minecraftforge.entity.IEntityAdditionalSpawnData;
+import net.minecraftforge.network.NetworkHooks;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import software.bernie.geckolib3.core.AnimationState;
+import software.bernie.geckolib3.core.IAnimatable;
+import software.bernie.geckolib3.core.PlayState;
+import software.bernie.geckolib3.core.builder.AnimationBuilder;
+import software.bernie.geckolib3.core.builder.ILoopType;
+import software.bernie.geckolib3.core.controller.AnimationController;
+import software.bernie.geckolib3.core.event.predicate.AnimationEvent;
+import software.bernie.geckolib3.core.manager.AnimationData;
+import software.bernie.geckolib3.core.manager.AnimationFactory;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+
+public class ServerCursorEntity extends LivingEntity implements IAnimatable, IEntityAdditionalSpawnData
```

Comment:
Whats the reason for using LivingEntity?

Reply:
Because IAnimatable can only be used on LivingEntity, not Entity

Reply:
Sorry I meant that _GeoEntityRender_ can only be used on a LivingEntity.